tests/integration/test_sqs.py

```diff
@@ -1398,16 +1400,631 @@ def test_batch_send_with_invalid_char_should_succeed(self, sqs_client, sqs_creat
+        queue_name = "queue-{}".format(short_uid())
+        queue_url = sqs_create_queue(QueueName=queue_name)
+
+        invalid_attribute = {
+            "attr1": {"StringValue": "Invalid-{},{}".format(chr(8), chr(11)), "DataType": "String"}
+        }
+
+        with pytest.raises(Exception) as e:
+            sqs_client.send_message(
+                QueueUrl=queue_url, MessageBody="test", MessageAttributes=invalid_attribute
+            )
+        e.match("InvalidParameterValue")
+
+    @pytest.mark.skip
+    def test_dead_letter_queue_execution_lambda_mapping_preserves_id(
+        self, sqs_client, sqs_create_queue, lambda_client, create_lambda_function
+    ):
+        # TODO: lambda triggered dead letter delivery does not preserve the message id
+        # https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-dead-letter-queues.html
+        queue_name = "queue-{}".format(short_uid())
+        dead_letter_queue_name = "dl-queue-{}".format(short_uid())
+        dl_queue_url = sqs_create_queue(QueueName=dead_letter_queue_name)
+
+        # create arn
+        url_parts = dl_queue_url.split("/")
+        region = os.environ.get("AWS_DEFAULT_REGION") or TEST_REGION
+        dl_target_arn = "arn:aws:sqs:{}:{}:{}".format(
+            region, url_parts[len(url_parts) - 2], url_parts[len(url_parts) - 1]
+        )
+
+        policy = {"deadLetterTargetArn": dl_target_arn, "maxReceiveCount": 1}
+        queue_url = sqs_create_queue(
+            QueueName=queue_name, Attributes={"RedrivePolicy": json.dumps(policy)}
+        )
+
+        lambda_name = "lambda-{}".format(short_uid())
+        create_lambda_function(
+            lambda_name,
+            libs=TEST_LAMBDA_LIBS,
+            handler_file=TEST_LAMBDA_PYTHON,
+            runtime=LAMBDA_RUNTIME_PYTHON36,
+        )
+        # create arn
+        url_parts = queue_url.split("/")
+        queue_arn = "arn:aws:sqs:{}:{}:{}".format(
+            region, url_parts[len(url_parts) - 2], url_parts[len(url_parts) - 1]
+        )
+        lambda_client.create_event_source_mapping(
+            EventSourceArn=queue_arn, FunctionName=lambda_name
+        )
+
+        # add message to SQS, which will trigger the Lambda, resulting in an error
+        payload = {lambda_integration.MSG_BODY_RAISE_ERROR_FLAG: 1}
+        result_send = sqs_client.send_message(QueueUrl=queue_url, MessageBody=json.dumps(payload))
+
+        assert poll_condition(
+            lambda: "Messages"
+            in sqs_client.receive_message(QueueUrl=dl_queue_url, VisibilityTimeout=0),
+            5.0,
+            1.0,
+        )
+        result_recv = sqs_client.receive_message(QueueUrl=dl_queue_url, VisibilityTimeout=0)
+        assert result_recv["Messages"][0]["MessageId"] == result_send["MessageId"]
+
 
 # TODO: test visibility timeout (with various ways to set them: queue attributes, receive parameter, update call)
 # TODO: test message attributes and message system attributes
```

Comment:
we have some tests her already right? what's still left TODO?

Reply:
system attribute tests seem pretty dry, wasn't sure if I should create completely new ones or imply refactor the existing ones